src/martialarts.cpp

```diff
@@ -848,7 +849,11 @@ bool martialart::has_technique( const Character &u, const matec_id &tec_id ) con
 
 bool martialart::has_weapon( const itype_id &itt ) const
 {
-    return weapons.count( itt ) > 0;
+    return weapons.count( itt ) > 0 ||
+           std::any_of( itt->weapon_category.begin(), itt->weapon_category.end(),
+    [&]( const std::string & weap ) {
+        return weapon_category.count( weap ) > 0;
+    } );
```

Comment:
This isn't the most efficient way to test for two sets intersecting, but a faster solution would be more verbose, which is probably not worthwhile for something non-performance-critical.

I *might* be inclined to separate this into a new `do_sets_intersect` function.  Makes me wonder if we already have similar code elsewhere.

Reply:
Having one would definitely be useful. I find myself checking if any member of a `set` is inside another `set` quite a lot.